tests/integration/snowflake/test_snowflake_adapter.py

```diff
@@ -324,3 +324,72 @@ def test_check_count_and_query(sf_adapter):
 
     assert exc.errisinstance(TooManyRecords)
     assert sf_adapter.check_count_and_query.retry.statistics["attempt_number"] == 4
+
+
+def test_generate_schema(sf_adapter):
+    DATABASE, SCHEMA = "SNOWSHU_DEVELOPMENT", "GENERATE_SCHEMA_TEST"
+    try:
+        # Clean up before test
+        if SCHEMA in sf_adapter._get_all_schemas(DATABASE):
+            sf_adapter.drop_schema(database=DATABASE, name=SCHEMA)
+
+        # Test schema creation
+        sf_adapter.generate_schema(database=DATABASE, name=SCHEMA)
+        assert SCHEMA in sf_adapter._get_all_schemas(database=DATABASE)
+    finally:
+        # Clean up after test
+        sf_adapter.drop_schema(database=DATABASE, name=SCHEMA)
+
+
+def test_drop_schema(sf_adapter):
+    DATABASE, SCHEMA = "SNOWSHU_DEVELOPMENT", "DROP_SCHEMA_TEST"
+    try:
+        # Setup: Ensure the schema exists before attempting to drop it
+        if SCHEMA not in sf_adapter._get_all_schemas(DATABASE):
+            sf_adapter.generate_schema(database=DATABASE, name=SCHEMA)
+
```

Comment:
one more assert to ensure that schema exists, before running the actual drop statement